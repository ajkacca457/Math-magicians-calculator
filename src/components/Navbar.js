import React from 'react';
import { Link } from 'react-router-dom';

const navstyle = {
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  backgroundColor: '#f5913e',
  margin: '0px',
  padding: '5px 10px',
};

const linkstyle = {
  textDecoration: 'none',
  color: 'white',
  marginRight: '10px',
};

const Navbar = () => (
  <div className="navbar" style={navstyle}>
    <h3 style={{ margin: '5px' }}>Math Magician Calculator</h3>
    <div className="links">
      <Link to="/home" style={linkstyle}>Home</Link>
      <Link to="/calculator" style={linkstyle}>Calculator</Link>
      <Link to="/quote" style={linkstyle}>Quote</Link>
    </div>

  </div>
);

export default Navbar;
