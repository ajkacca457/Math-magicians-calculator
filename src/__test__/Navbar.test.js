import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Navbar from '../components/Navbar';

describe('Navbar Component', () => {
  it('renders correctly', () => {
    const navtree = TestRenderer
      .create(<BrowserRouter>
        {' '}
        <Navbar />
        {' '}
              </BrowserRouter>)
      .toJSON();
    expect(navtree).toMatchSnapshot();
  });
});
