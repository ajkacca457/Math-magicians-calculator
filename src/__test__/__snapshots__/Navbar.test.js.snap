// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component renders correctly 1`] = `
Array [
  " ",
  <div
    className="navbar"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f5913e",
        "display": "flex",
        "justifyContent": "space-between",
        "margin": "0px",
        "padding": "5px 10px",
      }
    }
  >
    <h3
      style={
        Object {
          "margin": "5px",
        }
      }
    >
      Math Magician Calculator
      <i
        className="fas fa-magic"
        style={
          Object {
            "marginLeft": "5px",
          }
        }
      />
    </h3>
    <div
      className="links"
    >
      <a
        className="link"
        href="/home"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "marginRight": "15px",
            "textDecoration": "none",
          }
        }
      >
        <i
          className="fas fa-home"
          style={
            Object {
              "marginRight": "5px",
            }
          }
        />
        Home
      </a>
      <a
        className="link"
        href="/calculator"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "marginRight": "15px",
            "textDecoration": "none",
          }
        }
      >
        <i
          className="fas fa-calculator"
          style={
            Object {
              "marginRight": "5px",
            }
          }
        />
        Calculator
      </a>
      <a
        className="link"
        href="/quote"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "marginRight": "15px",
            "textDecoration": "none",
          }
        }
      >
        <i
          className="far fa-comments"
          style={
            Object {
              "marginRight": "5px",
            }
          }
        />
        Quote
      </a>
    </div>
  </div>,
  " ",
]
`;
